bounce/server/api/membership.py

```diff
@@ -65,30 +107,44 @@ class MembershipEndpoint(Endpoint):
     @validate(DeleteMembershipRequest, None)
     async def delete(self, request, club_name, id_from_token=None):
         """
-        Handles a DELETE /memberships/<club_name>?user_id=<user_id> request
+        Handles a DELETE /memberships/<club_name> request
         by deleting the membership that associates the given user with the
-        given club..
+        given club.
         """
-        # TODO: fix this when we have user roles set up. A user should only be
-        # able to delete their own memberships and memberships on clubs they
-        # are an admin/owner of.
 
         # Decode the club name
         club_name = unquote(club_name)
-        try:
-            user_id = int(request.args.get('user_id'))
-        except ValueError:
-            raise APIError('Invalid user ID', status=400)
-
-        if id_from_token != user_id:
-            # Regular members can only delete their own memberships
-            raise APIError('Forbidden', status=403)
 
         # Make sure the club exists
         club_row = club.select(self.server.db_session, club_name)
         if not club_row:
             raise APIError('No such club', status=404)
 
-        # Delete the memberships
-        membership.delete(self.server.db_session, club_name, user_id=user_id)
-        return response.text('', status=204)
+        try:
+            editor_attr = membership.select(self.server.db_session, club_name,
+                                            id_from_token, Roles.president)
+
+            editors_role = editor_attr[0]['role']
+
+            if 'user_id' in request.args:
+                user_id = int(request.args['user_id'])
+                # validate whether the user ID corresponds to an existing user
+                validate_user(self.server.db_session, user_id)
+                member_attr = membership.select(self.server.db_session,
+                                                club_name, user_id,
+                                                Roles.president)
+                members_role = member_attr[0]['role']
+            # Fetch the club's memberships
+            if 'user_id' in request.args:
```

Comment:
Do you need this extra `if` statement? It looks like you're already checking this condition above.

Reply:
got it.  thanks!